thunder/core/pytree.py

```diff
@@ -13,7 +16,31 @@
     namespace=optree.registry.__GLOBAL_NAMESPACE,
 )
 
-tree_flatten = partial(optree.tree_flatten, none_is_leaf=True)
+
+def tree_flatten(args):
+    if type(args) not in {
+        dict,
+        list,
+        str,
+        int,
+        bool,
+        tuple,
+        torch.dtype,
+        float,
+        dtypes.floating,
+        dtypes.bool_,
+        devices.Device,
+        torch.memory_format,
+        type(None),
+        slice,
+        complex,
+        type,
+        type(Ellipsis),
+    } and not isinstance(args, (ProxyInterface)):
```

Comment:
I think this should include `torch.Size` (which we register above for flattening) - otherwise it will start erroring.

Reply:
Thanks for catching this! Let me do that :)